java/Xpedite/src/com/xpedite/probes/AnchoredProbe.java

```diff
@@ -0,0 +1,37 @@
+///////////////////////////////////////////////////////////////////////////////
+//
+// Java implementation of Xpedite probes with line numbers
+//
+// Author: Brooke Elizabeth Cantwell, Morgan Stanley
+//
+///////////////////////////////////////////////////////////////////////////////
+
+package com.xpedite.probes;
+
+public class AnchoredProbe extends AbstractProbe {
+
+    private int lineNo;
+    private CallSite[] callSites = new CallSite[1];
+
+    private String buildProbeName(String className, String methodName, int lineNo) {
+        StringBuffer buffer = new StringBuffer(className);
+        buffer.append(".").append(methodName);
+        buffer.append(":").append(lineNo);
+        return buffer.toString();
+    }
+
+    public AnchoredProbe(String className, String methodName, int lineNo) {
+        super(className, methodName);
+        callSites[0] = new CallSite(this, buildProbeName(className, methodName, lineNo));
```

Comment:
`callSites =  new CallSite[1] { new CallSite(this, buildProbeName(className, methodName, lineNo))};`